File: 12.01.2026/valid_palindrome.java
import java.util.*;

public class ValidPalindrome {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.nextLine();

        String clean = s.replaceAll("[^a-zA-Z0-9]", "").toLowerCase();

        int left = 0, right = clean.length() - 1;
        boolean isPalindrome = true;

        while (left < right) {
            if (clean.charAt(left) != clean.charAt(right)) {
                isPalindrome = false;
                break;
            }
            left++;
            right--;
        }

        System.out.println(isPalindrome ? "Palindrome" : "Not Palindrome");
    }
}

File: 11.01.2026/valid_palindrome.py
s = input("Enter string: ")

clean = ""
for ch in s:
    if ch.isalnum():
        clean += ch.lower()

left, right = 0, len(clean) - 1
is_palindrome = True

while left < right:
    if clean[left] != clean[right]:
        is_palindrome = False
        break
    left += 1
    right -= 1

print("Valid Palindrome" if is_palindrome else "Not a Palindrome")
